src/geometry/rtree.hpp

```diff
@@ -75,8 +75,13 @@ class RTree {
   [[nodiscard]] auto lookup(doc_type const& doc) const -> boost::optional<geom_type const&>;
   void insert(geom_type const& geom, t_docId id);
 
+  // Predicte refers to the bgi::predicate concept that rtree.query(predicate) expects
```

Comment:
Do we have access to concepts in the c++ version we are using?
Maybe using a concept will better describe the predicate expected structure?

Reply:
@kei-nan Let's consider this in a separate PR. Such changes are better done on the master branch.
This is just a backport PR.